list_test.go

```diff
@@ -58,13 +70,87 @@ func TestList(t *testing.T) {
 	testDecodeFramework(t, "customReplyTypedVariableListHasNull", []interface{}{new(A0), new(A1), nil})
 	testDecodeFramework(t, "customReplyUntypedFixedListHasNull", []interface{}{new(A0), new(A1), nil})
 	testDecodeFramework(t, "customReplyUntypedVariableListHasNull", []interface{}{new(A0), new(A1), nil})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_A0", []*A0{new(A0), new(A0), nil})
+	testDecodeFramework(t, "customReplyTypedVariableList_A0", []*A0{new(A0), new(A0), nil})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_int", []int32{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_int", []int32{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_long", []int64{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_long", []int64{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_float", []float64{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_float", []float64{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_double", []float64{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_double", []float64{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_short", []int32{1, 2, 3})
+	testDecodeFramework(t, "customReplyTypedVariableList_short", []int32{1, 2, 3})
+
+	testDecodeFramework(t, "customReplyTypedFixedList_char", "123")
+	testDecodeFramework(t, "customReplyTypedVariableList_char", []string{"1", "2", "3"})
```

Comment:
Why not `"123"` instead of `[]string{"1", "2", "3"}`?

Reply:
![image](https://user-images.githubusercontent.com/18375310/60182151-498b8300-9856-11e9-8ae4-48bd3541e94c.png)
This will get []string{}

![image](https://user-images.githubusercontent.com/18375310/60182182-61fb9d80-9856-11e9-86a2-77deef069ef9.png)

This will get "123"

Reply:
But `customReplyTypedVariableList_char` is typed with `char`? Why is not same as `"123"`?

Reply:
I already change it to the first
.

Reply:
@fangyincheng Oh yes, it is really strange in java. So go hessian2 treats `char[]` as `string[]` because go doesn't have `char` type?

Reply:
`char[]` could be `[]byte` in golang

Reply:
@wongoo Why do I decode char[] to string[]?

Reply:
@fangyincheng  better to decode to `[]rune`, have a try